crates/wac-parser/src/resolution/package.rs

```diff
@@ -62,6 +62,9 @@ impl Package {
         version: Option<&Version>,
         bytes: Arc<Vec<u8>>,
     ) -> Result<Self> {
+        if !has_magic_header(&bytes) {
+            bail!("package '{name}' is expected to be a WebAssembly component binary but is not");
```

Comment:
Just a nit: I've been using backticks elsewhere to denote names.

Also, a small suggestion on the error message, but if you prefer yours instead, I'm happy to stick with it.

```suggestion
            bail!("package `{name}` is expected to be a binary WebAssembly component but it is missing the expected header");
```

Reply:
Switched to backticks and added "binary", but otherwise I think I'd personally prefer the original suggestion. Your suggestion is more precise, but I think that makes it less understandable. If the header is missing than it's almost guaranteed that some completely unrelated bag of bits is being passed and the most clear error IMO is just that wac has received a non-component binary.